src/main/webapp/app/tasks/default-task.service.ts

```diff
@@ -23,4 +23,9 @@ export class DefaultTaskService implements TaskService {
   getAll(): Observable<Task[]> {
     return this.http.get<Task[]>(this.baseUrl + '/tasks');
   }
+
+  search(searchText: string): Observable<Task[]> {
+    const querySearch =  searchText ? `?q=${searchText}` : '';
+    return this.http.get<Task[]>(`${this.baseUrl}/tasks${querySearch}`);
```

Comment:
did you want to add the `http://localhost:4200/?q=term` functionality by this?

Reply:
You're totally write, I didn't understand what is meant by ('deep-link'), I googled then I got it and I implemented this functionality. But I left this line because I think it's needed also for calling the API with search string. Is it right?